lib/logstash/inputs/http_poller.rb

```diff
@@ -42,6 +54,20 @@ def register
 
     @logger.info("Registering http_poller Input", :type => @type, :schedule => @schedule, :timeout => @timeout)
 
+    @url_field = ecs_select[disabled: "[#{metadata_target}][request][url]", v1: "[url][full]"]
+    @method_field = ecs_select[disabled: "[#{metadata_target}][request][method]", v1: "[http][request][method]"]
+    @host_field = ecs_select[disabled: "[#{metadata_target}][host]", v1: "[host][hostname]"]
+    @response_code_field = ecs_select[disabled: "[#{metadata_target}][code]", v1: "[http][response][status][code]"]
+    @response_headers_field = ecs_select[disabled: "[#{metadata_target}][response_headers]", v1: "[#{metadata_target}][input][http_poller][response][headers]"]
+    @response_message_field = ecs_select[disabled: "[#{metadata_target}][response_message]", v1: "[#{metadata_target}][input][http_poller][response][status][message]"]
+    @response_time_field = ecs_select[disabled: "[#{metadata_target}][runtime_seconds]", v1: "[#{metadata_target}][input][http_poller][response][time][second]"]
```

Comment:
believe we could leverage [`event.duration`](https://github.com/elastic/ecs/blob/1.11/generated/ecs/ecs_flat.yml#L2141) here (we would need to convert to nano-seconds), wdyt?

Reply:
happy to change it

Reply:
maybe in milliseconds ?

Reply:
yeah I know this gets tricky - we could "resolve" this with two fields (+ custom logic to convert to nanos):
```
    @response_time_field = ecs_select[disabled: "[#{metadata_target}][runtime_seconds]", v1: nil]
    @event_duration_field = ecs_select[disabled: nil, v1: "[event][duration]"]
```

Reply:
okay, fine by me. it's not a perfect fit but I was suggested using `event.duration` (from the ECS team) in a few places while doing the grok pattern updates. e.g. with HAProxy's *total_time_ms* which is kind of similar (and Beats uses `event.duration` in such cases despite the missing start/end fields) - the only reason we did not use it in the patterns was the need to convert to nanos: so I did not take the `duration` field description strictly.

but if the point is that the event is not "done yet" here it's a legit ...

Reply:
moving the discussion forward I think it would make more sense to have a request time fields that isn't in seconds.
since ECS uses nanosecond precision we should consider using that (or at least millis indicated by the field name).

e.g. `...[response][duration]`, `...[response][elapsed_time_ms]` or rather have it `...[request][duration]` prefixed?

Reply:
+1 `...[response][elapsed_time_ns]` prefer to keep it in nanoseconds as the same as event.duration

Reply:
:+1: we "could" strip the `_ns` suffix but I actually like that you've decided to keep it since the user looking at 
`elapsed_time_ns` field will have no doubt about the precision being used (despite nanos being default in ECS)